### PR TITLE
feat(profiling): Introduce additional continuous profiling feature flag

### PR DESCRIPTION
**Pull request by viglia**
Add logic to allow the ingestion of continuous profiles only for beta orgs if `continuous-profile-beta-ingest` feature is enabled.

This depends on:

- https://github.com/getsentry/sentry-options-automator/pull/2610
- https://github.com/getsentry/sentry/pull/80236

#skip-changelog

### CODE REVIEW

**Review by Dav1dde — CHANGES_REQUESTED** [review]
This logic of enabling/disabling features should be contained in Sentry, the feature handler can remove/add a feature flag also based on a time range (either for all or selected amount of projects), then Relay only needs to check for the feature flag.

**Comment by viglia** [comment]
> This logic of enabling/disabling features should be contained in Sentry, the feature handler can remove/add a feature flag also based on a time range (either for all or selected amount of projects), then Relay only needs to check for the feature flag.

@Dav1dde I have no problem moving this to a custom _FeatureHandler_ in _getsentry_, but I took this approach because I thought there was a push to move toward `flagpole` ([here ](https://www.notion.so/sentry/Flagpole-Spec-95238403e37a442f8af659a69e079e3d#579e6977791446b18505f0e109c16408) and [here ](https://www.notion.so/sentry/Flagpole-Spec-95238403e37a442f8af659a69e079e3d?pvs=4#ae29a1b5bdd34e8f99dc290b8942d21e))?

Are _FeatureHandler_ still the way to go in case of custom logic?

**Comment by Dav1dde** [comment]
> Are FeatureHandler still the way to go in case of custom logic?

As far as I am aware yes. 

I always understood flagpole as a replacement for flagr, but not any of the logic that builds upon the options framework in Sentry.

File: relay-dynamic-config/src/feature.rs
Comment (Dav1dde) [inline relay-dynamic-config/src/feature.rs]:
```suggestion
    /// Enable continuous profiling ingestion.
    ///
    /// Serialized as `organizations:continuous-profiling-allowed-ingest`.
```

File: relay-server/src/services/processor/profile_chunk.rs
Comment (Dav1dde) [inline relay-server/src/services/processor/profile_chunk.rs]: I wonder, can we instead of adding a new feature just use the existing `Feature::ContinuousProfiling` and either remove or add it in the feature handler?

File: relay-server/src/services/processor/profile_chunk.rs
Comment (Dav1dde) [inline relay-server/src/services/processor/profile_chunk.rs]:
```suggestion
    let continuous_profiling_enabled = state.project_info.has_feature(Feature::ContinuousProfiling)
        && state.project_info.has_feature(Feature::ContinuousProfilingAllowedIngest);
```

**Review by Dav1dde — COMMENTED** [review]
Wonder if we need a new feature flag or we can just use the existing one.

**Comment by jjbayer** [comment]
Agree with @Dav1dde here, best to keep the existing feature flag and determine its value in a feature handler.

File: relay-server/src/services/processor/profile_chunk.rs
Comment (viglia) [reply relay-server/src/services/processor/profile_chunk.rs]: > I wonder, can we instead of adding a new feature just use the existing `Feature::ContinuousProfiling` and either remove or add it in the feature handler?

@Dav1dde @jjbayer I agree it makes sense and indeed yesterday in the team weekly we discussed doing just that, but at the end I was asked not to mix the 2 things and just create a new one. So here we are.

**Review by Dav1dde — APPROVED** [review]

**Comment by Dav1dde** [comment]
Note when you merge this, it will disable all ingestion for continuous profile, maybe that is the desired outcome, if not the feature flag must be enabled before merging this.

**Comment by viglia** [comment]
@Zylphrex refactored as we discussed in the new rollout [plan](https://www.notion.so/sentry/Continuous-Profiling-Rollout-Extended-Beta-1358b10e4b5d8063bf42e07d08dd6237)

File: relay-dynamic-config/src/feature.rs
Comment (Zylphrex) [inline relay-dynamic-config/src/feature.rs]:
```suggestion
    /// Enabled when only beta orgs are allowed to send continuous profiles
```

File: relay-server/src/services/processor/profile_chunk.rs
Comment (Zylphrex) [inline relay-server/src/services/processor/profile_chunk.rs]: Something like this is probably easier to read
```suggestion
    let continuous_profiling_enabled = if state.project_info.has_feature(Feature::ContinuousProfilingBetaIngest) {
        state.project_info.has_feature(Feature::ContinuousProfilingBeta)
    } else {
        state.project_info.has_feature(Feature::ContinuousProfiling)
    };
```

**Review by Zylphrex — APPROVED** [review]
A few minor comments, but otherwise LGTM

**Review by jjbayer — CHANGES_REQUESTED** [review]
@viglia sorry for being pedantic here, but since this is temporary rollout logic and relay code has a longer life cycle than sentry's (customers running external relays), could you please move this logic to sentry-side project config generation?

If it cannot be done easily within the feature handler, you could add the custom logic to `get_exposed_features`.

**Review by jjbayer — APPROVED** [review]
Approving to unblock since this is temporary and external relays will still correctly fall back to the old feature flag.

**Pull request merged**
